### PR TITLE
Feat/add pico sdk

### PR DESCRIPTION
**Pull request by thewh1teagle**
Add pico sdk for Raspberry PI pico
inspired from arch linux package
https://aur.archlinux.org/packages/pico-sdk

You can test with
```console
git clone https://github.com/raspberrypi/pico-examples --depth 1
cd pico-examples
cmake -B build .
cmake --build build --target blink -- -j16
```

Used ninja as build system

### CODE REVIEW

**Comment by ognevnydemon** [comment]
you don't need to write install files for specific environments as they all are just the same

**Comment by thewh1teagle** [comment]
Fixed

**Comment by Biswa96** [comment]
Installation paths are incorrect. Please follow the PKGBUILD template files in this repository.

Also, it is not required to mention compiler in depends because this package is just header and cmake files.

**Comment by thewh1teagle** [comment]
@Biswa96 
The package description is
*Libraries and tools for C/C++ development on RP2040 microcontrollers.* (note the tools)
and in the package of arch linux it's install the compilers along with the sdk.

**Comment by Biswa96** [comment]
> and in the package of arch linux it's install the compilers along with the sdk too.

AUR != official archlinux packages. There are many headers only packages here in msys2 or in ArchLinux which you can refer.

**Comment by thewh1teagle** [comment]
@Biswa96 
Fixed

File: mingw-w64-pico-sdk/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Contributer: thewh1teagle
+_realname=pico-sdk
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:2]: Please specify your email and leave an empty line after first line

File: mingw-w64-pico-sdk/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=1.5.1
+pkgrel=1
+pkgdesc="Libraries for C/C++ development on RP2040 microcontrollers."
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:7]:
```suggestion
pkgdesc="Libraries for C/C++ development on RP2040 microcontrollers. (mingw-w64)"
```

File: mingw-w64-pico-sdk/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+arch=("any")
+mingw_arch=("mingw64" "ucrt64" "clang64")
+url="https://github.com/raspberrypi/pico-sdk"
+license=("BSD-3-Clause")
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:11]:
```suggestion
license=("spdx:BSD-3-Clause")
```

File: mingw-w64-pico-sdk/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+makedepends=("git")
+options=("!strip")
+install=${_realname}-${MSYSTEM}.install
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:14]:
```suggestion
install=${_realname}.install
```
One file should be sufficient. Keep just one and rename it.

File: mingw-w64-pico-sdk/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    # Clean SDK
+    rm -rf "${srcdir}/${_realname}/"{.git*,README.md,CONTRIBUTING.md,LICENSE.TXT}
+    
+    # Install SDK files
+    cp -r "${srcdir}/${_realname}" "${pkgdir}/share/${pkgname}"
+
+    # Install profile script
+    mkdir -p "${pkgdir}/etc/profile.d"
+    echo "export PICO_SDK_PATH=/share/${pkgname}" > "${pkgdir}/etc/profile.d/${_realname}.sh"
+}
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:39]:
```suggestion
}

```
Please add en empty line at the end.

File: mingw-w64-pico-sdk/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+sha256sums=("SKIP")
+
+prepare() {
+    cd "${srcdir}/${_realname}"
+    git submodule update --init --depth=1
+}
+
+package() {
+    # Install docs
+    install -Dm644 "${srcdir}/${_realname}/README.md" -t "${pkgdir}/share/doc/${pkgname}"
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:26]:
```suggestion
    install -Dm644 "${srcdir}/${_realname}/README.md" -t "${pkgdir}${MINGW_PREFIX}/share/doc/${pkgname}"
```
replace /usr with ${MINGW_PREFIX}

File: mingw-w64-pico-sdk/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+    # Install license
+    install -Dm644 "${srcdir}/${_realname}/LICENSE.TXT" -t "${pkgdir}/share/licenses/${pkgname}"
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:29]:
```suggestion
    install -Dm644 "${srcdir}/${_realname}/LICENSE.TXT" -t "${pkgdir}${MINGW_PREFIX}/share/licenses/${pkgname}"
```

File: mingw-w64-pico-sdk/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+    # Clean SDK
+    rm -rf "${srcdir}/${_realname}/"{.git*,README.md,CONTRIBUTING.md,LICENSE.TXT}
+    
+    # Install SDK files
+    cp -r "${srcdir}/${_realname}" "${pkgdir}/share/${pkgname}"
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:35]:
```suggestion
    cp -r "${srcdir}/${_realname}" "${pkgdir}${MINGW_PREFIX}/share/${pkgname}"
```

File: mingw-w64-pico-sdk/PKGBUILD [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+    # Install profile script
+    mkdir -p "${pkgdir}/etc/profile.d"
+    echo "export PICO_SDK_PATH=/share/${pkgname}" > "${pkgdir}/etc/profile.d/${_realname}.sh"
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:39]:
```suggestion
    mkdir -p "${pkgdir}${MINGW_PREFIX}/etc/profile.d"
    echo "export PICO_SDK_PATH=/share/${pkgname}" > "${pkgdir}${MINGW_PREFIX}/etc/profile.d/${_realname}.sh"
```

File: mingw-w64-pico-sdk/pico-sdk.install [hunk 10]
```diff
@@ -0,0 +1,8 @@
+post_install() {
+	tput bold
+	echo "To apply the changes, please either:"
+	echo "1. Relaunch the terminal"
+	echo "   or"
+	echo "2. Execute: source /etc/profile"
```
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/pico-sdk.install:6]:
```suggestion
	echo "2. Execute: source ${MINGW_PREFIX}/etc/profile"
```

File: mingw-w64-pico-sdk/PKGBUILD (hunk 9 above)
Comment (thewh1teagle) [reply mingw-w64-pico-sdk/PKGBUILD:39]: then the export path should include `MINGW_PREFIX` too

Comment (MehdiChinoune) [reply mingw-w64-pico-sdk/PKGBUILD:39]: yes

File: mingw-w64-pico-sdk/pico-sdk.install (hunk 10 above)
Comment (thewh1teagle) [reply mingw-w64-pico-sdk/pico-sdk.install:6]: Looks like it doesn't exists
```console
$ stat $MINGW_PREFIX/etc/profile
stat: cannot stat '/ucrt64/etc/profile': No such file or directory
```

Comment (MehdiChinoune) [reply mingw-w64-pico-sdk/pico-sdk.install:6]: It's profile.d

File: mingw-w64-pico-sdk/PKGBUILD (hunk 6 above)
Comment (MehdiChinoune) [inline mingw-w64-pico-sdk/PKGBUILD:26]:
```suggestion
    install -Dm644 "${srcdir}/${_realname}/README.md" -t "${pkgdir}/share/doc/${_realname}"
```

File: mingw-w64-pico-sdk/pico-sdk.install (hunk 10 above)
Comment (thewh1teagle) [reply mingw-w64-pico-sdk/pico-sdk.install:6]: how about
```console
source "${MINGW_PREFIX}/etc/profile.d/${_realname}.sh"
```
Not sure if this variables available in that post install script

**Comment by ognevnydemon** [comment]
> Fixed

I didn't mean that! I meant to .install files. very sorry, I missed it

**Comment by thewh1teagle** [comment]
@ognevnydemon for some reason I understood it incorrectly.
Anyway, hope the current version sufficient

**Comment by Alexpux** [comment]
@thewh1teagle @Biswa96 I understand that there are packages with headers only but they are usable for mingw targets. What reason to add this package? We dont provide toolchain to work with it, so what it give to users? If you just want to package headers to have it on your own you can just grab it from repo

**Comment by thewh1teagle** [comment]
@Alexpux 
Mingw do provide compiler to work with 
`arm-none-eabi-gcc`

Also, we can add toolchain package too.

It gives to users A LOT by providing fast and easy way to setup environment for developing for raspberry pi. Again, it gives A LOT. Try to see what is the current way users does that (check it on youtube and see how complicated process it is)
So as any other package, it simplify the process. 

How it looks like without mingw:
User buy raspberry pi pico, and setup the environment with 6 different softwares he install,
4 times adding to PATH environment
And very heavy tools which takes long time to download because they not packaged efficiently as mongw do.

How it looks like with mingw?
User buy raspberry, install mingw, open the terminal and install the tools in one command, 2 minutes of installation and that's it!

**Comment by Alexpux** [comment]
@thewh1teagle **MINGW** packages repo is for **NATIVE** windows targets software mostly. We dont provide cross toolchain **arm-none-eabi-gcc** here. If you want to have it you need create cross toochain **msys->arm** in MSYS packages or create separate repository for arm cross toolchain and add packages there.

**Comment by thewh1teagle** [comment]
You do have packages for arm development. 
And here is gcc for arm:
https://packages.msys2.org/package/mingw-w64-ucrt-x86_64-arm-none-eabi-gcc?repo=ucrt64

**Comment by Alexpux** [comment]
@thewh1teagle I see now. Maybe better to have separate repo for cross tools? I see cross GCC builded in 2022 year and not updated since this time. Is it still work without rebuilding for current GCC?

**Comment by thewh1teagle** [comment]
It works. 
What do you mean saparte repo?

**Comment by Alexpux** [comment]
@thewh1teagle no matter. I mean create separate repository for cross toolchains

**Comment by thewh1teagle** [comment]
@Alexpux 
separate repository can be "Separate msys2 repository for install `pico-sdk` with `pacman`"
or just "Separate `github` repository" to guide users install manually.

**Comment by Alexpux** [comment]
@thewh1teagle need to discuss about it in discord maybe. I mean create separate github repo for cross toolchain packages with its own install prefix, like **/crosstools** or something else. So it will save size and time to build it once, not for all /{clang,mingw} prefixes

**Comment by Biswa96** [comment]
I am not sure whether this package should be added.

**Comment by thewh1teagle** [comment]
Close for inactivity. 
If you still think to merge it just reopen / merge.

And please, before reviewing packages, let the contributors know if you gonna merge it, 
before you tell them to do extra work.

**Pull request closed without merging**